pkg/kubectl/cmd/portforward.go

```diff
@@ -68,81 +90,102 @@ func NewCmdPortForward(f *cmdutil.Factory, cmdOut, cmdErr io.Writer) *cobra.Comm
 }
 
 type portForwarder interface {
-	ForwardPorts(method string, url *url.URL, config *restclient.Config, ports []string, stopChan <-chan struct{}) error
+	ForwardPorts(method string, url *url.URL, opts PortForwardOptions) error
 }
 
 type defaultPortForwarder struct {
 	cmdOut, cmdErr io.Writer
 }
 
-func (f *defaultPortForwarder) ForwardPorts(method string, url *url.URL, config *restclient.Config, ports []string, stopChan <-chan struct{}) error {
-	dialer, err := remotecommand.NewExecutor(config, method, url)
+func (f *defaultPortForwarder) ForwardPorts(method string, url *url.URL, opts PortForwardOptions) error {
+	dialer, err := remotecommand.NewExecutor(opts.Config, method, url)
 	if err != nil {
 		return err
 	}
-	fw, err := portforward.New(dialer, ports, stopChan, f.cmdOut, f.cmdErr)
+	fw, err := portforward.New(dialer, opts.Ports, opts.StopChannel, f.cmdOut, f.cmdErr)
 	if err != nil {
 		return err
 	}
+	opts.ReadyChannel = fw.Ready
```

Comment:
Shouldn't this assignment be reversed? You want to override the [default portforward ready channel](https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/portforward/portforward.go#L123) with whatever the caller provided in PortForwardOptions, not overwrite the options.